main.py

```diff
@@ -59,3 +64,191 @@ async def request_classification(request: Request):
             "classification_scores": json.dumps(classification_scores),
         },
     )
+
+
+
+@app.get("/custom_classifications")
+def create_classify(request: Request):
+    return templates.TemplateResponse(
+        "custom_classification_select.html",
+        {"request": request, "models": Configuration.models},
+    )
+
+
+@app.post("/custom_classifications")
+async def upload_file(file: UploadFile, request: Request):
+    try:
+        # Legge il contenuto del file
+        file_content = await file.read()
+
+        # Salva il file nel server nella cartella "uploads" --> imagenet_subset
+        original_path = f"app/static/imagenet_subset/{file.filename}"
+        with open(original_path, "wb") as f:
+            f.write(file_content)
+
+        # Controlla il formato dell'immagine e converte se necessario
+        if not file.filename.upper().endswith('.JPEG'):
+            file_name_without_extension = file.filename.rsplit('.', 1)[0]
+            converted_path = f"app/static/imagenet_subset/{file_name_without_extension}.JPEG"
+
+            with Image.open(original_path) as img:
+                img.convert("RGB").save(converted_path, "JPEG")
+
+            os.remove(original_path)  # Rimuove la vecchia copia
+
+            # Aggiorna il nome del file con l'estensione .JPEG
+            image_id = f"{file_name_without_extension}.JPEG"
+        else:
+            # Se il file è già in formato .JPEG, usa il nome originario
+            image_id = file.filename
+
+        form = ClassificationForm(request)
+        await form.load_data()
+        model_id = form.model_id
+        classification_scores = classify_image(model_id=model_id, img_id=image_id)
+
+        return templates.TemplateResponse(
+            "classification_output.html",
+            {
+                "request": request,
+                "image_id": image_id,
+                "classification_scores": json.dumps(classification_scores),
+            },
+        )
+    except Exception as e:
+        return {"error": f"Si è verificato un errore durante l'upload del file: {str(e)}"}
```

Comment:
The uploaded files are always saved in the `static` folder and never removed. Even if a file is not a valid image. This should be avoided as files may accumulate without any control of their content and memory usage.

Reply:
Additional suggestion (you don't have to implement it): it's not safe to check the file format by only looking at its extension. A safer method is to check the magic number, for instance with [python-magic](https://github.com/ahupp/python-magic).

Reply:
We have solved the problem with the upload of images, with the current code the image is temporarily saved in the static/imagenet_subset folder so that it can be used by the classifier. Once it is finished using it, it is removed, and displayed in the html page through the use of a buffer. 
Now the python-magic library is used for a check on the format of the file, so you can see if this is an image or not (this library is added to requirements.txt).

Reply:
Thank you! Note that in the current state, some critical issues remain:
1. if a user uploads a file with the same name as a pre-loaded image, it will be overwritten (and deleted after the classification);
2. if a user uploads a file with the same name as the labels JSON file, it will be overwritten and the application will stop working correctly;
3. if multiple users upload files with the same name, the app may behave unpredictably.

Reply:
Sorry, I forgot to update the PR description, I have now added the links to the related issues

Reply:
No problem. I edited a bit the PR description to automatically close all the issues when merging.